src/visualization.jl

```diff
@@ -78,127 +79,166 @@ using DocStringExtensions
 import DiffEqBase: DiscreteCallback, ODESolution, CallbackSet, u_modified!, terminate!
 import DataStructures
-import RigidBodyDynamics: MechanismState, normalize_configuration!
-import LoopThrottle: @throttle
+import RigidBodyDynamics: MechanismState, normalize_configuration!, configuration
+using Observables: Observable
+using InteractBase: Widget, button, observe
+using WebIO: Node, Scope
+using Blink: Window, body!
+
+struct SimulationStatus
+    terminate::Observable{Bool}
+    pause::Observable{Bool}
+end
+
+function initialize!(commands::SimulationStatus)
+    commands.terminate[] = false
+    commands.pause[] = false
+end
 
 const DEFAULT_PAUSE_POLLINT = 0.05
 
-function CommandHandler(commands::SimulationCommands; pause_pollint::Float64 = DEFAULT_PAUSE_POLLINT)
+function CommandHandler(status::SimulationStatus; pause_pollint::Float64 = DEFAULT_PAUSE_POLLINT)
     condition = (u, t, integrator) -> true
-    action = let commands = commands
+    action = let status = status
         function (integrator)
             yield()
-            while commands.pause && !commands.terminate
+            while status.pause[] && !status.terminate[]
                 sleep(pause_pollint)
             end
-            commands.terminate && (commands.terminate = false; terminate!(integrator))
+            status.terminate[] && (status.terminate[] = false; terminate!(integrator))
             u_modified!(integrator, false)
         end
     end
-    initialize = (c, t, u, integrator) -> VisualizerInterface.initialize!(commands)
+    initialize = (c, t, u, integrator) -> VisualizerInterface.initialize!(status)
     DiscreteCallback(condition, action, initialize = initialize, save_positions=(false, false))
 end
 
-function TransformPublisher(state::MechanismState, vis; max_fps = 60.)
+function TransformPublisher(vis::MechanismVisualizer; max_fps = 60.)
     last_update_time = Ref(-Inf)
     condition = let last_update_time = last_update_time, min_Δt = 1 / max_fps
         function (u, t, integrator)
             last_time_step = length(integrator.opts.tstops) == 1 && t == DataStructures.top(integrator.opts.tstops)
             last_time_step || time() - last_update_time[] >= min_Δt
         end
     end
-    action = let state = state, vis = vis, last_update_time = last_update_time
+    action = let vis = vis, last_update_time = last_update_time
         function (integrator)
             last_update_time[] = time()
-            copy!(state, integrator.u)
-            visualize(vis, integrator.t, state)
+            copy!(vis, integrator.u)
             u_modified!(integrator, false)
         end
     end
     DiscreteCallback(condition, action; save_positions = (false, false))
 end
 
+@deprecate CallbackSet(vis, state::MechanismState; max_fps = 60.) CallbackSet(vis, state; max_fps = max_fps)
+
+struct SimulationControls{W <: Widget}
+    terminate::W
+    pause::W
+end
+
+SimulationControls() = SimulationControls(button("terminate"), button("pause"))
```

Comment:
Is there a way to get the pause button to stay depressed when the simulation is paused?